src/components/Input/SketchCanvas.js

```diff
@@ -0,0 +1,237 @@
+/* eslint-disable import/no-anonymous-default-export */
+import React, { useEffect, useRef, useState } from 'react'
+import { ReactSketchCanvas } from 'react-sketch-canvas'
+import { SketchPicker } from 'react-color'
+import trashcanIcon from '../../assets/icons/trashcan.svg'
+import colorpickerIcon from '../../assets/icons/colorpicker.svg'
+import undoIcon from '../../assets/icons/undo.svg'
+import redoIcon from '../../assets/icons/redo.svg'
+import { mergeRefs } from '../../utility/utilities'
+import styled from 'styled-components'
+
+const CanvasContainer = styled.div`
+  background: #f3f3f3;
+  display: flex;
+  flex-direction: column;
+  position: relative;
+`
+
+const ErrorMessage = styled.p`
+  position: absolute;
+  top: 0px;
+  left: 0px;
+  padding: 5px;
+  padding-left: 8px;
+  padding-right: 8px;
+  margin: 0px;
+  background: #ffaaaa;
+  color: red;
+  z-index: 10;
+  box-sizing: border-box;
+`
+
+const Stroke4Button = styled.div`
+  background: #333333;
+  border-radius: 999px;
+  cursor: pointer;
+  width: 8px;
+  height: 8px;
+`
+
+const Stroke6Button = styled.div`
+  background: #333333;
+  border-radius: 999px;
+  cursor: pointer;
+  width: 12px;
+  height: 12px;
+`
+
+const Stroke8Button = styled.div`
+  background: #333333;
+  border-radius: 999px;
+  cursor: pointer;
+  width: 16px;
+  height: 16px;
+`
+
+const CanvasIcon = styled.img`
+  width: 20px;
+  height: 25px;
+  cursor: pointer;
+`
+
+const HistoryIcons = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 5px;
+  align-items: center;
+`
+
+const ColorPickerWrapper = styled.div`
+  position: absolute;
+  z-index: 100;
+  top: 0;
+  right: 0;
+`
+
+const LeftIcons = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  gap: 5px;
+`
+
+const ToolBar = styled.div`
+  display: flex;
+  flex-direction: row;
+  padding: 3px;
+  padding-left: 7px;
+  padding-right: 7px;
+  justify-content: space-between;
+  user-select: none;
+`
+/**
+ * @param width number, width of the entire component
+ * @param height number, height of the entire component
+ * @param invalid boolean, indicates dissatisfaction with current value
+ * @param errorMsg string, indicates some error
+ * @param onChange (result: string) => void, triggers with result (SVG or Base64) on change. Has debouncing.
+ * @param customRef a ref directly to the canvas component
```

Comment:
i believe customRef is for storing the redirect hash? like if there is a required field, and the applicant doesn't fill it out, it will scroll up to that customRef

Reply:
done? I believe lincoln added better documentation